src/ngx_http_lua_balancer.c

```diff
@@ -299,7 +299,28 @@ ngx_http_lua_balancer_get_peer(ngx_peer_connection_t *pc, void *data)
     if (bp->sockaddr && bp->socklen) {
         pc->sockaddr = bp->sockaddr;
         pc->socklen = bp->socklen;
-        pc->name = &bp->host;
+
+        /* return a copy pointer of bp->host to pc, in case upstream_addr
+         * always use the last peer name
+         */
+        ngx_str_t *peer_name = ngx_palloc(r->pool, sizeof(ngx_str_t));
+        if (peer_name == NULL) {
+            pc->name = &bp->host;
+            ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+                "failed to malloc mem from request pool, upstream addr may not correct");
+        } else {
+            peer_name->data = ngx_palloc(r->pool, bp->host.len);
+            if (peer_name->data == NULL) {
+                pc->name = &bp->host;
+                ngx_log_error(NGX_LOG_ERR, r->connection->log, 0,
+                    "failed to malloc mem from request pool, upstream addr may not correct");
+            } else {
```

Comment:
Style: need a blank line before `} else {`. Please check the rest of the code base for the right coding style. Thank you.